File: BOJ/1926.py
# 그림

from collections import deque


def bfs(x, y):
    dq = deque([[x, y]])
    visited[x][y] = 1
    acc = 1
    while dq:
        row, col = dq.popleft()
        for i in range(4):
            nx = row + dx[i]
            ny = col + dy[i]
            if nx < 0 or ny < 0 or nx >= n or ny >= m:
                continue
            if visited[nx][ny]:
                continue
            if board[nx][ny]:
                dq.append([nx, ny])
                acc += 1
                visited[nx][ny] = 1
    values.append(acc)


n, m = map(int, input().split())
board = [list(map(int, input().split())) for _ in range(n)]
visited = [[0] * m for _ in range(n)]
count = 0
values = []

dx = [-1, 0, 1, 0]
dy = [0, 1, 0, -1]

for i in range(n):
    for j in range(m):
        if board[i][j] and not visited[i][j]:
            count += 1
            bfs(i, j)

print(count)
if values:
    print(max(values))
else:
    print(0)

File: BOJ/silver/1926.js
// 1926 - 그림

const fs = require("fs");
const file = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const inputs = fs
  .readFileSync(file)
  .toString()
  .trim()
  .split("\n")
  .map((v) => v.split(" ").map(Number));
const [n, m] = inputs[0];
const board = inputs.slice(1);

const solution = (n, m, board) => {
  const visited = Array.from({ length: n }, () => Array(m).fill(0));
  const dy = [-1, 0, 1, 0];
  const dx = [0, 1, 0, -1];
  const result = [];

  const bfs = (startY, startX) => {
    const queue = [];

    let size = 1;

    queue.push([startY, startX]);
    visited[startY][startX] = 1;

    while (queue.length) {
      const [y, x] = queue.shift();

      for (let dir = 0; dir < 4; dir++) {
        const ny = y + dy[dir];
        const nx = x + dx[dir];

        if (ny < 0 || ny >= n || nx < 0 || nx >= m) continue;
        if (visited[ny][nx] || board[ny][nx] !== 1) continue;

        queue.push([ny, nx]);
        visited[ny][nx] = 1;
        size++;
      }
    }

    return size;
  };

  for (let i = 0; i < n; i++) {
    for (let j = 0; j < m; j++) {
      if (visited[i][j] === 0 && board[i][j] !== 0) {
        result.push(bfs(i, j));
      }
    }
  }

  const answer = [result.length, result.length ? Math.max(...result) : 0];

  return answer.join("\n");
};

console.log(solution(n, m, board));
